shared/src/question-generators/propositional-logic/minimize.ts

```diff
@@ -0,0 +1,189 @@
+    freetext_feedback_parse_error: "Deine Antwort ist kein gültiger aussagenlogischer Ausdruck.",
+    freetext_feedback_no_normal_form: "Deine Antwort ist keine {{0}}.",
+    freetext_feedback_not_equivalent: "Deine Antwort ist **nicht** äquivalent zu $\\varPhi$.",
+    freetext_feedback_not_minimal:
+      "Deine Antwort is äquivalent zu $\\varPhi$, aber sie ist nicht minimal.",
+    DNF: "DNF",
+    CNF: "KNF",
+  },
+}
+
+/**
+ * Question generator for "minimize a boolean expression"
+ */
+export const MinimizePropositionalLogic: QuestionGenerator = {
+  id: "plminimize",
+  name: tFunctional(translations, "name"),
+  description: tFunctional(translations, "description"),
+  tags: ["boolean logic", "propositional logic", "propositional calculus", "normal forms", "CNF", "DNF"],
+  languages: ["en", "de"],
+  expectedParameters: [
+    {
+      name: "variant",
+      type: "string",
+      allowedValues: ["dnf"], // "CNF"
+    },
+  ],
+
+  generate: (lang = "en", parameters, seed) => {
+    const path = serializeGeneratorCall({
+      generator: MinimizePropositionalLogic,
+      lang,
+      parameters,
+      seed,
+    })
+    const random = new Random(seed)
+
+    const variant: "DNF" = (parameters.variant as "dnf").toUpperCase() as "DNF"
+    const numVariables = random.int(3, 4)
+    const numLeaves = random.int(numVariables + 2, numVariables + 4)
+    const varNames = random.choice(variableNames).slice(0, numVariables)
+
+    let randExpression
+    do {
+      randExpression = generateRandomExpression(random, numLeaves, varNames)
+    } while (!randExpression.getProperties().falsifiable || !randExpression.getProperties().satisfiable)
+
+    const typingsAids = getTypingAids(lang)
+    const question: FreeTextQuestion = {
+      type: "FreeTextQuestion",
+      name: MinimizePropositionalLogic.name(lang),
+      path,
+      prompt: "$\\varPhi^*=$",
+      text: t(translations, lang, "text", [
+        randExpression.toString(true),
+        t(translations, lang, variant),
+      ]),
+      feedback: feedbackFunction(randExpression, variant, lang),
+      checkFormat: checkFormat(varNames),
+      typingAid: [
+        typingsAids.leftParenthesis,
+        typingsAids.rightParenthesis,
+        typingsAids.logicalOr,
+        typingsAids.logicalAnd,
+        typingsAids.logicalNot,
+      ].concat(getTypingAidsVars(lang, randExpression.getVariableNames()).typingAidVars),
+    }
+
+    return { question }
+  },
+}
+
+/**
+ * Simple function to parse the user input
+ * Catch ParsingError and show, it's not parsable
+ * @param varNames
+ */
+function checkFormat(varNames: string[]): FreeTextFormatFunction {
+  return ({ text }) => {
+    const res = PropositionalLogicParser.parse(text)
+
+    return {
+      valid: !(res instanceof ParserError || _.difference(res.getVariableNames(), varNames).length > 0),
+      message: `$ ${tokenToLatex(text)}$`,
```

Comment:
![Screenshot From 2024-11-11 15-17-33](https://github.com/user-attachments/assets/69775528-cee7-478e-9eb9-80ecb6a01131)
some operands don't get displayed correctly, though the first one of each type seems to generally work.

Reply:
Yes, it's a known problem. I ll fix it on `main` 👍